### PR TITLE
Add exchangeratesapi.io data source - Closes #468

### LINKED ISSUE #468: Add exchangeratesapi.io data source
### Description

Create a job that retrieves data from http://exchangeratesapi.io/.

These jobs, registered during start, take care of regular database updates with source data, using data from 3rd party services. These updates are performed during regular intervals, according to the usage policy written in the terms of service.

The job should be identified as `prices.retrieve.exchangeratesapi`.

Convert the data to the format described in additional information section.

Store retrieved data in Redis with its update date.

Refer to the table in additional information to get more details about the source endpoint.

### Motivation

A reliable data source is required to keep market prices up-to-date.

### Acceptance Criteria

- Data is retrieved, converted and stored in the Redis database.
- Update intervals do not exceed terms of service.
- Update date is stored in the database.

### Additional Information

Source name | exchangeratesapi.io
-- | --
Category | FIAT
Data source | European Central Bank
Availability | Free, unlimited
Limits | Unknown
Open Source | Yes
Format | JSON
Docs | https://exchangeratesapi.io/
ToS/AGB | MIT
Access URL | https://api.exchangeratesapi.io/latest

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #468 

### How was it solved?
- [x] Add job `prices.retrieve.exchangeratesapi` to retrieve data from `exchangeratesapi.io` api
- [x] Transform data according to the requirement
- [x] Store transformed data from `exchangeratesapi.io` into `redis` cache
- [x] Add function `getExchangeratesapiPricesFromDB` to retrieve latest prices from `exchangeratesapi.io` cache
- [x] Added `ttl` of 24 hours and make request only when data is expired

### How was it tested?
- [x] Local
- [x] Jenkins

### CODE REVIEW

File: services/market/config.js [hunk 1]
```diff
@@ -47,4 +55,8 @@ config.log.file = process.env.SERVICE_LOG_FILE || 'false';
 // Set docker host if running inside the container
 config.log.docker_host = process.env.DOCKER_HOST || 'local';
 
+// Api keys to access apis
+config.access_key = {};
+config.access_key.exchangeratesapi = 'e27c981843d2d08916cf4e3864f5a92b';
```
Comment (sameersubudhi) [inline services/market/config.js:60]: Please consider passing on sensitive information such as `access keys` via environment variables.

File: services/market/shared/market/sources/exchangeratesapi.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+const fetchAllMarketTickers = async () => {
+    try {
+        const allMarketTickers = {};
+        await BluebirdPromise.all(
+            currencies.map(async (baseCurrency) => {
+                const response = await requestLib(`${apiEndpoint}/latest?access_key=${accessKey}&base=${baseCurrency}&symbols=${currencies.join(',')}`);
+                if (response) allMarketTickers[baseCurrency] = response.data.rates;
+            }),
+        );
+        return allMarketTickers;
```
Comment (sameersubudhi) [inline services/market/shared/market/sources/exchangeratesapi.js:47]: As a good to have, please consider removing the `baseCurrency` from the `symbols` list as the relationship would always be a constant equal to `1`.

Also, please consider renaming the methods/variables to be more appropriate and reflective of their purpose.

```suggestion
const fetchAllCurrencyConversionRates = async () => {
    try {
        const allMarketConversionRates = {};
        await BluebirdPromise.all(
            currencies.map(async (baseCurrency) => {
                const remainingCurrencies = currencies.filter(c => c !== baseCurrency);
                const response = await requestLib(`${apiEndpoint}/latest?access_key=${accessKey}&base=${baseCurrency}&symbols=${remainingCurrencies.join(',')}`);
                if (response) allMarketConversionRates[baseCurrency] = response.data.rates;
            }),
        );
        return allMarketConversionRates;
```

File: services/market/shared/market/sources/exchangeratesapi.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            from,
+            to,
+            rate: ticker.price,
+            updateTimestamp: Math.floor(Date.now() / 1000),
+            sources: ['exchangeratesapi'],
+        };
+        return price;
+    });
+    return transformedPrices;
+};
```
Comment (sameersubudhi) [inline services/market/shared/market/sources/exchangeratesapi.js:72]: Same as above. Please consider renaming `prices` to `conversionRates` instead!?

File: services/market/shared/market/sources/exchangeratesapi.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    // Check if prices exists in cache
+    if (!pricesFromCache) {
+        const tickers = await fetchAllMarketTickers();
+        const transformedPrices = standardizeTickers(tickers);
+
+        // Serialize individual price item and write to the cache
+        await BluebirdPromise.all(transformedPrices
+            .map(item => exchangeratesapiCache.set(`exchangeratesapi_${item.code}`, JSON.stringify(item), 24 * 60 * 60 * 1000)));
+    }
+};
```
Comment (sameersubudhi) [inline services/market/shared/market/sources/exchangeratesapi.js:101]: Same as above.
Also, please consider adding `ttl` value as a top-level file constant or an entry in the config file.

```suggestion
const reloadPricesFromExchangerateapi = async () => {
    const conversionRatesFromCache = await getExchangeratesapiPricesFromDB();

    // Check if prices exists in cache
    if (!conversionRatesFromCache) {
        const currencyConversionRates = await fetchAllCurrencyConversionRates();
        const transformedRates = standardizeCurrencyConversionRates(currencyConversionRates);

        // Serialize individual price item and write to the cache
        await BluebirdPromise.all(transformedRates
            .map(item => exchangeratesapiCache.set(`exchangeratesapi_${item.code}`, JSON.stringify(item), 24 * 60 * 60 * 1000)));
    }
};
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please find a few observations/recommendations below.

**Review by sameersubudhi — APPROVED** [review]
Looks good. Thank you 👍

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**